pragma-sdk/pragma_sdk/common/fetchers/generic_fetchers/deribit.py

```diff
@@ -0,0 +1,304 @@
+        instrument_name_felt = str_to_felt(self.instrument_name)
+        base_currency_felt = str_to_felt(self.base_currency)
+        return compute_hash_on_elements(  # type: ignore[no-any-return]
+            [
+                instrument_name_felt,
+                base_currency_felt,
+                self.current_timestamp,
+                int(self.mark_price),
+            ]
+        )
+
+
+CurrenciesOptions = Dict[str, List[OptionData]]
+
+
+@dataclass
+class LatestData:
+    merkle_tree: MerkleTree
+    options: CurrenciesOptions
+
+
+class DeribitOptionsFetcher(FetcherInterfaceT):
+    """
+    Deribit fetcher.
+    Retrieves all the options data for all available instruments for a set of pairs.
+
+    All the available prices are put in a Merkle tree which is sent in a GenericEntry.
+    """
+
+    publisher: str
+    pairs: List[Pair]
+    headers: Dict[Any, Any]
+    _client: PragmaOnChainClient
+    _latest_data: Optional[LatestData] = None
+
+    REQUIRED_PAIRS = {
+        Pair.from_tickers("BTC", "USD"),
+        Pair.from_tickers("ETH", "USD"),
+    }
+
+    SOURCE: str = "DERIBIT"
+    BASE_URL: str = "https://www.deribit.com/api/v2/public"
+    ENDPOINT_OPTIONS: str = "get_book_summary_by_currency?currency="
+    ENDPOINT_OPTIONS_SUFFIX: str = "&kind=option&expired=false"
+
+    def __init__(
+        self,
+        pairs: List[Pair],
+        publisher: str,
+        api_key: Optional[str] = None,
+        network: Network = "mainnet",
+    ):
+        super().__init__(pairs, publisher, api_key, network)
+        self._latest_data = None
+        if set(self.pairs) != self.REQUIRED_PAIRS:
+            raise ValueError(
+                "Currently, DeribitOptionsFetcher must be used for BTC/USD and ETH/USD only."
+            )
+
+    async def fetch(
+        self, session: ClientSession
+    ) -> List[Entry | PublisherFetchError | BaseException]:
+        """
+        For every unique base assets in the provided pairs list, fetch all the
+        available instruments and their option data.
+        They'll be merged in a merkle tree, and we only return the merkle root
+        using the GenericEntry type (so we only return one Entry in the final list).
+        """
+        currencies_options: CurrenciesOptions = {}
+
+        unique_currencies = list(set([pair.base_currency for pair in self.pairs]))
+        for currency in unique_currencies:
+            currencies_options[currency.id] = await self._fetch_options(
+                session, currency
+            )
+
+        merkle_tree: MerkleTree = self._build_merkle_tree(currencies_options)
+        self._set_latest_data(merkle_tree, currencies_options)
+
+        entry = GenericEntry(
+            key=DERIBIT_MERKLE_FEED_KEY,
+            value=merkle_tree.root_hash,
+            timestamp=int(time.time()),
+            source=str_to_felt(self.SOURCE),
+            publisher=str_to_felt(self.publisher),
+        )
+        return [entry]
```

Comment:
```suggestion
        return self._construct(merkle_tree)
```
to align with other fetchers

Reply:
done